### PR TITLE
ANDROID-13687 Improve Snackbar accessibility

### PR DESCRIPTION
**Pull request by jeslat**
### :goal_net: What's the goal?
Improve Snackbar accessibility.

**Old**  
~~The snackbar will get the focus when it is opened.~~

**New** 
The snackbar should no get the focus when it is shown, it will have focus in the default A11y order.
The snackbar should announce the message following this logic:
- Informative: should announce the message without interrupt the current Talkback announcement.
- Critical: should announce the message interrupting the current Talkback announcement.

The main action will be spoken as a button with an optional custom content description.

The idea was not to introduce a breaking change with the content description, but after trying this version in android-messenger, for example in Java code, it is a breaking change.

### :construction: How do we do it?
* Provide options to add a custom content description to the main action.
* Get accessibility focus when the snackbar is opened.
* Change main action from `TextView` to `Button` to improve the talkback message.

### ☑️ Checks
- [x] I updated the documentation, including readmes and wikis. If this is a breaking change, tag the PR with "Breaking Change" label and remember to include breaking change migration guide in release notes where this version is released.

### :test_tube: How can I test this?
- Video (i have added some delay before the Snackbar display to be able to check the interruption):

https://github.com/Telefonica/mistica-android/assets/51969539/08b6e214-7e97-4aa3-b73d-db4d249559c6

- After changing the TextView to a Button, I've verified that the snackbar maintains the same UI. You can verify with these screenshots the result:

| Before        | After           |
| ------------- |-------------|
| drawing      | drawing |
| drawing      | drawing |
| drawing      | drawing |
| drawing      | drawing |
| drawing      | drawing |
| drawing      | drawing |

- [x] Reviewed by Mistica design team

### CODE REVIEW

File: library/src/main/java/com/telefonica/mistica/feedback/SnackbarBuilder.kt [hunk 1]
```diff
@@ -107,6 +110,11 @@ open class SnackbarBuilder(view: View?, text: String) {
         snackbar.setCustomAction()
         snackbar.showDismissActionIfNeeded(hasInfiniteDuration = snackbarLength == SnackbarLength.INDEFINITE)
         snackbar.addCallbackIfNeeded()
+        snackbar.addCallback(object : BaseCallback<Snackbar>() {
+            override fun onShown(transientBottomBar: Snackbar) {
+                transientBottomBar.view.sendAccessibilityEvent(AccessibilityEvent.TYPE_VIEW_FOCUSED)
+            }
+        })
```
Comment (jeslat) [inline library/src/main/java/com/telefonica/mistica/feedback/SnackbarBuilder.kt:117]: This is to get the focus when it's opened

File: library/src/main/java/com/telefonica/mistica/feedback/snackbar/CustomSnackbarLayout.kt [hunk 2]
```diff
@@ -79,15 +83,17 @@ internal class CustomSnackbarLayout @JvmOverloads constructor(
         constraintSet.connect(action.id, ConstraintSet.TOP, text.id, ConstraintSet.BOTTOM)
         constraintSet.connect(action.id, ConstraintSet.END, parent.id, ConstraintSet.END)
         constraintSet.setHorizontalBias(action.id, 1.0F)
-        constraintSet.setMargin(action.id, ConstraintSet.END, context.resources.getDimensionPixelSize(R.dimen.mistica_snackbar_padding_horizontal))
+        constraintSet.setMargin(action.id, ConstraintSet.END, context.resources.getDimensionPixelSize(R.dimen.mistica_snackbar_action_horizontal_margin))
+        constraintSet.setMargin(action.id, ConstraintSet.BOTTOM, context.resources.getDimensionPixelSize(R.dimen.mistica_snackbar_action_vertical_margin))
+        constraintSet.setMargin(action.id, ConstraintSet.TOP, context.resources.getDimensionPixelSize(R.dimen.mistica_snackbar_action_vertical_margin))
```
Comment (jeslat) [inline library/src/main/java/com/telefonica/mistica/feedback/snackbar/CustomSnackbarLayout.kt:88]: I needed to adjust a bit the layout after changing to a Button

**Review by juangardi21 — APPROVED** [review]
LGTM 👍🏼

**Review by pmartinbTEF — APPROVED** [review]
Good job!

**Review by pmartinbTEF — APPROVED** [review]
LGTM

**Review by juangardi21 — APPROVED** [review]
LGTM 🚀

**Comment by aweell** [comment]
LGTM, I have a question left: 

What happens when is persistent snackbar displayed and a sheet is opened? the snackbar would remain under the sheet?

**Comment by jeslat** [comment]
> LGTM, I have a question left:
> 
> What happens when is a persistent snackbar displayed and a sheet is opened? the snackbar would remain under the sheet?

@aweell right, the snack bar remains under the sheet. You can see it in this video:

https://github.com/Telefonica/mistica-android/assets/2582348/7df71428-1d82-4367-aa14-5d7dd2794450

**Review by aweell — APPROVED** [review]

**Pull request merged**